qemu/tests/multi_vms_nics.py

```diff
@@ -114,45 +112,30 @@ def file_transfer(session, src, dst):
         if md5sum1 != md5sum2:
             raise error.TestError("File changed after transfer")
 
-    vm_list = []
-    session_list = []
+    ip_list = []
     vms = params["vms"].split()
     timeout = float(params.get("login_timeout", 360))
-    int_ip_filter = params["interface_ip_filter"]
     strict_check = params.get("strick_check", "no")
     host_ip = utils_net.get_ip_address_by_interface(params.get("netdst"))
     host_ip = params.get("srchost", host_ip)
     flood_minutes = float(params["flood_minutes"])
+    ips = []
     for vm_name in vms:
+        guest_ifname = ""
+        guest_ip = ""
         vm = env.get_vm(vm_name)
-        vm_list.append(vm)
-        session_list.append(vm.wait_for_login(timeout=timeout))
-
-    ip_list = []
-
-    error.context("Check all the nics available or not", logging.info)
-    count_nics = len(params.get("nics").split())
-    for i in session_list:
-        ips = []
-        cmd = params.get("net_check_cmd")
-        end_time = time.time() + timeout
-        while time.time() < end_time:
-            status, output = i.get_command_status_output(cmd)
-            if status:
-                err_msg = "Can not get ip from guest."
-                err_msg += " Cmd '%s' fail with output: %s" % (cmd, output)
-                logging.error(err_msg)
-            ips = re.findall(int_ip_filter, output, re.S)
-            if count_nics == len(ips):
-                break
-            time.sleep(2)
-        else:
-            err_log = "Not all nics get ip.  Set '%s' nics." % count_nics
-            err_log += " Guest only get '%s' ip(s). " % len(ips)
-            err_log += " Command '%s' output in guest:\n%s" % (cmd, output)
+        session = vm.wait_for_login(timeout=timeout)
+        error.context("Check all the nics available or not", logging.info)
+        guest_ifname = utils_net.get_linux_ifname(session,
+                                                  vm.get_mac_address())
+        guest_ip = vm.get_address()
+        if not (guest_ifname and guest_ip):
+            err_log = "vms %s get ip or ifname failed." % vm_name
+            err_log = "ifname: %s, ip: %s." % (guest_ifname, guest_ip)
             raise error.TestFail(err_log)
-        for ip in ips:
-            ip_list.append(ip + (i,))
+        ips = [guest_ifname, guest_ip, session]
```

Comment:
@luckyh thanks for your comment. have updated the patch. could you help to review it again?